main.rb

```diff
@@ -163,13 +163,19 @@ def appstore_version
   bundle_id = env_has_key('AC_BUNDLE_ID')
   country = get_env('AC_APPSTORE_COUNTRY')
   uri = if country
-          URI("http://itunes.apple.com/lookup?bundleId=#{bundle_id}&country=#{country}")
+          puts "Fetching version from App Store for Bundle Identifier: #{bundle_id} and country: #{country}"
+          URI("https://itunes.apple.com/lookup?bundleId=#{bundle_id}&country=#{country}")
         else
-          URI("http://itunes.apple.com/lookup?bundleId=#{bundle_id}")
+          puts "Fetching version from App Store for Bundle Identifier: #{bundle_id}"
+          URI("https://itunes.apple.com/lookup?bundleId=#{bundle_id}")
         end
   response = Net::HTTP.get_response(uri)
-  abort("Error: Received an unexpected status code from the iTunes Search API for bundle ID '#{bundle_id}' and country '#{country}'.".red) unless response.is_a?(Net::HTTPSuccess)
+  abort("Error: Received an unexpected status code from the iTunes Search API for Bundle Identifier '#{bundle_id}' and country '#{country}'. Please ensure the Bundle Identifier is correct and the app is available in the App Store or the specified country.".red) unless response.is_a?(Net::HTTPSuccess)
   response_body = JSON.parse(response.body)
+
+  if response_body['resultCount'].to_i == 0 || response_body['results'].nil? || response_body['results'].empty?
```

Comment:
(Nice to have) You can shorten the line length of the if condition like this:
```suggestion
  results = response_body['results'].to_a
  if response_body['resultCount'].to_i == 0 || results.empty?
```